clippy_lints/src/types.rs

```diff
@@ -531,6 +553,17 @@ fn span_precision_loss_lint(cx: &LateContext, expr: &Expr, cast_from: Ty, cast_t
     );
 }
 
+fn span_lossless_lint(cx: &LateContext, expr: &Expr, op: &Expr, cast_from: Ty, cast_to: Ty) {
+    span_lint_and_sugg(cx,
+                       CAST_LOSSLESS,
+                       expr.span,
+                       &format!("casting {} to {} may become silently lossy if types change",
```

Comment:
Perhaps we should add a `note: from() functions can only be lossless, but as conversions are lossy`

Reply:
This is covered in the linked wiki section, so my preference is to leave it at that. However, I can add a note if you think that's better.